src/NetTopologySuite/Operation/Linemerge/LineSequencer.cs

```diff
@@ -480,7 +480,11 @@ private Geometry BuildSequencedGeometry(IEnumerable<IEnumerable<DirectedEdge>> s
 
         private static LineString Reverse(LineString line)
         {
-            var pts = line.Coordinates;
+            var coordinates = line.Coordinates;
```

Comment:
If we'd copy anything, then we'd copy the `CoordinateSequence`.
Gee, why doesn't his whole function work on `CoordinateSeqeuence` in the first place?

Reply:
In fact, `CoordinateSequence` has a `.Reverse()` method that is documented to always return a reversed copy of the original sequence:
```csharp
var rev = line.Factory.CreateLineString(line.CoordinateSequence.Reverse());
```